### PR TITLE
Add support to validate both docker and ECR as image source 

### PR DESCRIPTION
**Pull request by Divyaasm**
### Description
Enhance docker validation too validate both dockerhub and ecr at a time

### Issues Resolved
https://github.com/opensearch-project/opensearch-build/issues/4756

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: jenkins/validate-artifacts/validate-artifacts.jenkinsfile [hunk 1]
```diff
@@ -97,8 +97,8 @@ pipeline {
         )
         choice(
             name: 'DOCKER_SOURCE',
-            description: 'Enter the docker-source to pull the docker image, either DockerHub or ECR.',
-            choices: ["dockerhub", "ecr"]
+            description: 'Enter the docker-source to pull the docker image, either DockerHub or ECR or Both.',
+            choices: ["dockerhub", "ecr", "Both"]
```
Comment (peterzhuamazon) [inline jenkins/validate-artifacts/validate-artifacts.jenkinsfile:101]: Why capital?

File: jenkins/validate-artifacts/validate-artifacts.jenkinsfile [hunk 2]
```diff
@@ -131,6 +131,10 @@ pipeline {
                         PROJECTS = "opensearch opensearch-dashboards"
                     }
 
+                    if (DOCKER_SOURCE == 'Both') {
+                        DOCKER_SOURCE = "dockerhub ecr"
+                    }
```
Comment (peterzhuamazon) [inline jenkins/validate-artifacts/validate-artifacts.jenkinsfile:136]: If this is the case I would rather we give an option of :
```
            choices: ["dockerhub", "ecr", "dockerhub ecr"]
```

File: src/run_validation.py [hunk 3]
```diff
@@ -21,8 +21,16 @@ def main() -> int:
...
-        logging.info(f'final test_result = {test_result}')
+        if args.distribution == "docker":
+            docker_source = args.docker_source
+            for source in docker_source:
+                docker_args = args
+                docker_args.docker_source = source
+                test_result = ValidationTestRunner.dispatch(docker_args, args.distribution, work_dir).run()
+
+        else:
+            test_result = ValidationTestRunner.dispatch(args, args.distribution, work_dir).run()
```
Comment (peterzhuamazon) [inline src/run_validation.py:32]: Why this change?

File: jenkins/validate-artifacts/validate-artifacts.jenkinsfile (hunk 1 above)
Comment (Divyaasm) [reply jenkins/validate-artifacts/validate-artifacts.jenkinsfile:101]: It aligns with existing implementation https://github.com/opensearch-project/opensearch-build/blob/6508fcd795e2e0998ff760356f47bfcfc1bee776/jenkins/validate-artifacts/validate-artifacts.jenkinsfile#L103C8-L107C10

File: jenkins/validate-artifacts/validate-artifacts.jenkinsfile (hunk 2 above)
Comment (Divyaasm) [reply jenkins/validate-artifacts/validate-artifacts.jenkinsfile:136]: Same as above ^^

File: src/run_validation.py (hunk 3 above)
Comment (Divyaasm) [reply src/run_validation.py:32]: It calls docker validation independently based on the values in --docker-source.

Comment (peterzhuamazon) [reply src/run_validation.py:32]: I see. Now you can call multiple of that.
But wont that be an issue if a previous dispatch already occupy same ports?
I assume you are not binding ports to the system right?

File: jenkins/validate-artifacts/validate-artifacts.jenkinsfile (hunk 1 above)
Comment (peterzhuamazon) [reply jenkins/validate-artifacts/validate-artifacts.jenkinsfile:101]: Got it.

File: jenkins/validate-artifacts/validate-artifacts.jenkinsfile (hunk 2 above)
Comment (peterzhuamazon) [reply jenkins/validate-artifacts/validate-artifacts.jenkinsfile:136]: Got it.

File: src/run_validation.py (hunk 3 above)
Comment (Divyaasm) [reply src/run_validation.py:32]: No, tested the changes, working as expected. The second pull takes place only after the cleanup. How ever it's mostly used by the release-promotion job.

Comment (peterzhuamazon) [reply src/run_validation.py:32]: After offline discussion, dispatch is not async but rather wait for completion.

Thanks.

**Review by peterzhuamazon — APPROVED** [review]
